contracts/masset/liquidator/Liquidator.sol

```diff
@@ -119,9 +119,16 @@ contract Liquidator is ILiquidator, Initializable, ModuleKeysStorage, ImmutableM
      * @notice Liquidator approves Uniswap to transfer Aave, COMP and ALCX tokens
      * @dev to be called via the proxy proposeUpgrade function, not the constructor.
      */
-    function upgrade() external {
+    function initialize() external initializer {
         IERC20(aaveToken).safeApprove(address(uniswapRouter), type(uint256).max);
         IERC20(compToken).safeApprove(address(uniswapRouter), type(uint256).max);
+    }
+
+    /**
+     * @notice Liquidator approves Uniswap to transfer Aave, COMP and ALCX tokens
+     * @dev to be called via the proxy proposeUpgrade function, not the constructor.
+     */
+    function upgrade() external {
```

Comment:
Not convinced of the need to store `alchemixToken` as an immutable, but also a minor point

Reply:
It's needed because the Liquidator has a proxy. The proxy needs to give the approval to Uniswap to spend the ALCX tokens.
I can't move to the `initialize` function either as the existing proxy has already given the approval for AAVE and COMP to be spent by Umiswap. And as they are safeApproves, the will fail if run again.